src/bp/core/services/auth/auth-service.ts

```diff
@@ -47,15 +50,31 @@ export default class AuthService {
   }
 
   async checkUserAuth(email: string, password: string, newPassword?: string, ipAddress: string = '') {
-    const user = await this.findUserByEmail(email || '', ['email', 'password', 'salt', 'password_expired'])
+    const user = await this.findUserByEmail(email || '', [
+      'email',
+      'password',
+      'salt',
+      'password_expired',
+      'password_expiry_date',
+      'unsuccessful_logins',
+      'locked_out'
+    ])
 
     if (!user || !validateHash(password || '', user.password!, user.salt!)) {
       this.stats.track('auth', 'login', 'fail')
       this.logger.info(`Login failed. User "${email}" from IP "${ipAddress}"`)
+
+      user && (await this._incrementWrongPassword(user))
       throw new InvalidCredentialsError()
     }
 
-    if (user.password_expired && !newPassword) {
+    if (user.locked_out) {
+      this.logger.info(`Login failed. User "${email}" from IP "${ipAddress}" is locked out`)
```

Comment:
I'd not use info but rather `warn` or `DEBUG()` ?  What we should ultimately do is have an express middleware that prints the custom errors to the console with their respective error code

Reply:
Good point on debug, i've moved other audits as well (login failed, etc)